### ISSUE TITLE
kldivergence of Categorical Dist (no method matching quantile with ::Int64)

### ISSUE THREAD

**Issue by JobJob** [JuliaStats/Distributions.jl#704]
```
kldivergence(Categorical([0.1, 0.1, 0.8]), Categorical([0.2, 0.3, 0.5]))
```
```
MethodError: no method matching quantile(::Distributions.Categorical{Float64}, ::Int64)
Closest candidates are:
  quantile(::Distributions.Hypergeometric, ::Real) at ...
  ...
```

I guess [quantile for Categorical](https://github.com/JuliaStats/Distributions.jl/blob/master/src/univariate/discrete/categorical.jl#L171
) should probably accept `p::Real` rather than `p::Float64`.

Looking at the history of the relevant files I'm guessing this never worked?

**Comment by GordStephen** [comment]
I think this is fixed now that #634 is merger?

**Comment by JobJob** [comment]
Yeah it works, thanks! I have a new issue though...
